code/modules/food_and_drinks/kitchen_machinery/cooker.dm

```diff
@@ -111,6 +113,32 @@
 		to_chat(user, "<span class='warning'>Close the panel first!</span>")
 		return
 	if(!checkValid(I, user))
+		if(istype(I, /obj/item/weapon/grab))
+			var/obj/item/weapon/grab/G = I
+			if(!iscarbon(G.affecting))
+				to_chat(user, "<span class='warning'>You can only harm carbon-based creatures this way!</span>")
```

Comment:
Fast response. We can add a secondary check to prevent against IPCs (though one imagines it'd burn their circuitry).